### ISSUE TITLE
[CKEditor 5] Upcoming editor v18.0.0 release has breaking changes that will crash MathType

### ISSUE THREAD

**Issue by mlewand** [wiris/html-integrations#21]
The upcoming CKEditor 5 release contains a [change in style processor](https://github.com/ckeditor/ckeditor5/issues/6091) that will break the MathType plugin. It causes instant crash with:

```plaintext
mathtype.js:66 TypeError: Cannot read property 'stylesProcessor' of undefined
    at new h (element.js:114)
    at b.domToView (domconverter.js:430)
    at b.domChildrenToView (domconverter.js:466)
    at domChildrenToView.next (<anonymous>)
    at b.domToView (domconverter.js:445)
    at o.toView (htmldataprocessor.js:79)
    at plugin.js:383
    at view (plugin.js:370)
    at xt.e.on.priority (downcasthelpers.js:654)
    at xt.fire (emittermixin.js:209)
```

The breaking changes are listed in the ["StylesProcessor is no longer singleton" pull request](https://github.com/ckeditor/ckeditor5-engine/pull/1826#issue-377772902).

**Comment by mlewand** [comment]
We went though your code and compiled a list of changes required to make it work, here it is:

*   [ps://github.com/wiris/html-integrations/blob/stable/packages/mathtype-ckeditor5/src/plugin.js#L242](https://github.com/wiris/html-integrations/blob/stable/packages/mathtype-ckeditor5/src/plugin.js#L242) – you shouldn’t use the ViewDocFrag constructor (it’s protected), but writer.createDocumentFragment(). You need to change it because we added new param to the constructor (the writer method didn’t change).
*   [https://github.com/wiris/html-integrations/blob/stable/packages/mathtype-ckeditor5/src/plugin.js#L239](https://github.com/wiris/html-integrations/blob/stable/packages/mathtype-ckeditor5/src/plugin.js#L239) – the constructor now gets the view document (see nightly docs here: [https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/api/module_engine_dataprocessor_xmldataprocessor-XmlDataProcessor.html#function-constructor](https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/api/module_engine_dataprocessor_xmldataprocessor-XmlDataProcessor.html#function-constructor))
*   [https://github.com/wiris/html-integrations/blob/stable/packages/mathtype-ckeditor5/src/plugin.js#L350](https://github.com/wiris/html-integrations/blob/stable/packages/mathtype-ckeditor5/src/plugin.js#L350) – same here.
*   [https://github.com/wiris/html-integrations/blob/stable/packages/mathtype-ckeditor5/src/plugin.js#L379](https://github.com/wiris/html-integrations/blob/stable/packages/mathtype-ckeditor5/src/plugin.js#L379) – and here (nightly docs: [https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/api/module_engine_dataprocessor_htmldataprocessor-HtmlDataProcessor.html](https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/api/module_engine_dataprocessor_htmldataprocessor-HtmlDataProcessor.html))
*   [https://github.com/wiris/html-integrations/blob/stable/packages/mathtype-ckeditor5/src/integration.js#L171](https://github.com/wiris/html-integrations/blob/stable/packages/mathtype-ckeditor5/src/integration.js#L171) – use `model.createPositionBefore()` [https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/api/module_engine_model_model-Model.html#function-createPositionBefore](https://ckeditor5.github.io/docs/nightly/ckeditor5/latest/api/module_engine_model_model-Model.html#function-createPositionBefore)

It's possible that we missed something, but this should do it. We based this on the `stable` branch.

**Comment by xaviripo** [comment]
Solved as of [MathType for CKEditor 5 version 7.19.0](https://www.npmjs.com/package/@wiris/mathtype-ckeditor5/v/7.19.0).
